pkg/trace/sampler/catalog.go

```diff
@@ -70,19 +70,22 @@ func (cat *serviceKeyCatalog) register(svcSig ServiceSignature) Signature {
 
 // ratesByService returns a map of service signatures mapping to the rates identified using
 // the signatures.
-func (cat *serviceKeyCatalog) ratesByService(rates map[Signature]float64, totalScore float64) map[ServiceSignature]float64 {
-	rbs := make(map[ServiceSignature]float64, len(rates)+1)
+func (cat *serviceKeyCatalog) ratesByService(localRates, remoteRates map[Signature]float64, defaultRate float64) map[ServiceSignature]float64 {
+	rbs := make(map[ServiceSignature]float64, len(localRates)+1)
 	cat.mu.Lock()
 	defer cat.mu.Unlock()
 	for key, el := range cat.items {
 		sig := el.Value.(catalogEntry).sig
-		if rate, ok := rates[sig]; ok {
+		// todo:raphael distinguish remote rates from local rates with a separate priority value
+		if rate, ok := remoteRates[sig]; ok {
```

Comment:
But this way the tracer won't know the difference between one and the other. Isn't that needed to add priority rate 3? How do you plan on fixing this? I was thinking you might need something like:
```json
{
    "rates_by_service": { ... },
    "remote_rates": { ... },
}
```

Reply:
Will do that next release, this one is a first iteration allowing us to test rates propagation.